include/envoy/http/header_map.h

```diff
@@ -269,6 +269,7 @@ class HeaderEntry {
   HEADER_FUNC(AccessControlAllowCredentials)                                                       \
   HEADER_FUNC(AccessControlExposeHeaders)                                                          \
   HEADER_FUNC(AccessControlMaxAge)                                                                 \
+  HEADER_FUNC(Age)                                                                                 \
```

Comment:
My feeling on this PR is to not include these headers into the header map right now, as doing so has a perf implication, and we don't have a production ready filter that uses them. Can you instead just use the O(N) header functions for now with these headers and we can put in a TODO to revisit once https://github.com/envoyproxy/envoy/issues/8795 is solved? cc @jmarantz @antoniovicente 

Assuming we agree on this I would just close this PR and merge this one into whatever PR actually uses these headers.

Reply:
RE header-map improvements, @adisuissa is looking at that, so hopefully this will become cheap in the near future.

RE granularity of PR: it's hard to get this exactly right, but it makes sense to just include in the PR that uses them, as this is pretty small. At some point you have to break things up but let's see the filter PR first.

Reply:
Will do.